File: 3626-smallest-divisible-digit-product-i/3626-smallest-divisible-digit-product-i.go
func smallestNumber(n int, t int) int {
    for {
        num := n
        pro := 1
        for  num > 0 {
            pro *= (num % 10)
            num /= 10
        }
        if pro % t == 0 {
            return n
        }
        n++
    }
    return -1
}

File: 3626-smallest-divisible-digit-product-i/3626-smallest-divisible-digit-product-i.cpp
class Solution {
public:
    int smallestNumber(int n, int t) {
        for (; n; ++n) {
            int num = n;
            int pro = 1;
            while (num) {
                pro *= (num % 10);
                num /= 10;
            }
            if (pro % t == 0)
                return n;
        }
        return -1;
    }
};